Use Variable Size Indices in Sparse Qlinear Code

salilsdesai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85247
* #85246
* #85245
* #85244
* #85243

Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

salilsdesai committed:
Use Variable Size Indices in Sparse Qlinear Code

Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]

digantdesai reviewed (approved):
LGTM, thanks @salilsdesai

salilsdesai committed:
Update on "Use Variable Size Indices in Sparse Qlinear Code"


Final changes to enable sparse weight packing with variable size indices

pack_block_sparse.cc is deleted because all functions in it have a template added, so they are moved to pack_block_sparse.h

Differential Revision: [D39025651](https://our.internmc.facebook.com/intern/diff/D39025651/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D39025651/)!

[ghstack-poisoned]
